### ISSUE TITLE
Labels not removed in any mode if text is edited before jumping the cursor

### ISSUE THREAD

**Issue by rohantalesara** [ice9js/ace-jump-sublime#69]
In any mode, if before choosing a label to jump to, the text content is edited, and then the jump is made, the labels don't get removed even after the jump has finished.
If the file is now saved, the labels get saved instead of the actual text content.
The following are screen shots detailing the same:

![ace-1](https://user-images.githubusercontent.com/31321664/31436989-5fdb1846-aea1-11e7-8080-f91f961eaafc.png)

![ace-2](https://user-images.githubusercontent.com/31321664/31437009-6d3341da-aea1-11e7-8727-a487f9d57248.png)

The labels remain even after the jump has finished:
![ace-3](https://user-images.githubusercontent.com/31321664/31437019-739375ae-aea1-11e7-828b-3e59ce6515a5.PNG)

After saving the file:
![ace-4](https://user-images.githubusercontent.com/31321664/31437026-7b2fc93e-aea1-11e7-8ccd-42d2c61b792e.PNG)

**Comment by RohanBS** [comment]
Pressing (Ctrl+Z) before saving removes the labels. However , if the file is saved without doing so , the labels do gets saved instead of the actual text content.

**Comment by ice9js** [comment]
Yes. Unfortunately the way it works is that AceJump is actually editing the files to add the labels, and then undoing that change once you make the jump. If you make any change in between - it will undo that instead of removing the labels.  
Do you think setting the buffers to `read-only` while the labels are shown would be a sufficient solution ?
